From e506e06d55e5200dd9159d68632c8e2b3152f64d Mon Sep 17 00:00:00 2001
From: Tuukka Toivonen <tuukka.toivonen@intel.com>
Date: Fri, 28 Nov 2014 13:58:40 +0200
Subject: [PATCH 158/164] atomisp2: Remove long delay in resuming from d3

AtomISP is not a true PCI device on SOC, so we can ignore the 10ms
delay when resuming from d3.

Change-Id: I194d01a8c7d424ea7af34a4d511251a551f77b70
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-10419
Signed-off-by: Tuukka Toivonen <tuukka.toivonen@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/30576
Reviewed-by: Jukka O Kaartinen <jukka.o.kaartinen@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../pci/atomisp2/atomisp_driver/atomisp_v4l2.c     |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
index fd50305..f7bee07 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_v4l2.c
@@ -1204,6 +1204,9 @@ static int atomisp_pci_probe(struct pci_dev *dev,
 	mutex_init(&isp->streamoff_mutex);
 	spin_lock_init(&isp->lock);
 
+	/* This is not a true PCI device on SoC, so the delay is not needed. */
+	isp->pdev->d3_delay = 0;
+
 	isp->media_dev.driver_version = ATOMISP_CSS_VERSION_21;
 	switch (id->device & ATOMISP_PCI_DEVICE_SOC_MASK) {
 	case ATOMISP_PCI_DEVICE_SOC_MRFLD:
-- 
1.7.9.5

